src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -41,11 +43,24 @@ class TaskStatusUpdateProcessorImpl @Inject() (
   private[this] val newKillUnknownTaskTimeMetric: Timer =
     metrics.timer("debug.killing-unknown-task-duration")
 
+  private[this] val taskStateCounterMetrics: mutable.Map[Int, Counter] = mutable.Map()
+
+  private[this] def getTaskStateCounterMetric(taskState: MesosProtos.TaskState): Counter = {
+    def createCounter() = {
+      val stateName = taskState.name().toLowerCase(Locale.US).replace('_', '-')
+      val metricName = s"mesos.task-states.$stateName"
+      metrics.counter(metricName)
+    }
+    taskStateCounterMetrics.getOrElseUpdate(taskState.getNumber, createCounter)
+  }
+
   logger.info("Started status update processor")
 
   override def publish(status: MesosProtos.TaskStatus): Future[Unit] = oldPublishTimeMetric {
     newPublishTimeMetric {
       logger.debug(s"Received status update\n${status}")
+      getTaskStateCounterMetric(status.getState).increment()
```

Comment:
can't this be called from multiple threads? If so, then the update of the metrics map is not thread safe...

But looking at the `MarathonScheduler` it should process only one callback at a time so we should be fine.

Do we maybe want to mention the fact that it's safe because of that somewhere?